Sources/_OpenAPIGeneratorCore/Parser/YamsParser.swift

```diff
@@ -58,9 +65,63 @@ struct YamsParser: ParserProtocol {
             throw OpenAPIVersionError(versionString: "openapi: \(openAPIVersion)")
         }
 
-        return try decoder.decode(
-            OpenAPI.Document.self,
-            from: input.contents
-        )
+        do {
+            return try decoder.decode(
+                OpenAPI.Document.self,
+                from: input.contents
+            )
+        } catch DecodingError.dataCorrupted(let errorContext) {
+            try possibly_throw_parsing_error(context: errorContext, input: input, diagnostics: diagnostics)
+            throw DecodingError.dataCorrupted(errorContext)
+        }
+    }
+
+    /// Detect specific YAML parsing errors to emit nicely formatted errors for IDEs
+    private func possibly_throw_parsing_error(
+        context: DecodingError.Context,
+        input: InMemoryInputFile,
+        diagnostics: DiagnosticCollector
+    ) throws {
+        if let yamlError = context.underlyingError as? YamlError {
+            if case .parser(let yamlContext, let yamlProblem, let yamlMark, _) = yamlError {
+                try throw_parsing_error(
+                    context: yamlContext,
+                    problem: yamlProblem,
+                    lineNumber: yamlMark.line - 1,
+                    input: input,
+                    diagnostics
+                )
+            } else if case .scanner(let yamlContext, let yamlProblem, let yamlMark, _) = yamlError {
+                try throw_parsing_error(
+                    context: yamlContext,
+                    problem: yamlProblem,
+                    lineNumber: yamlMark.line - 1,
+                    input: input,
+                    diagnostics
+                )
+            }
+        } else if let openAPIError = context.underlyingError as? OpenAPIError {
+            var problem = Diagnostic.error(message: openAPIError.localizedDescription)
+            problem.absoluteFilePath = input.absolutePath
+            diagnostics.emit(problem)
+            throw ExitCode.failure
+        }
+    }
+
+    private func throw_parsing_error(
```

Comment:
```suggestion
    private func throwParsingError(
```

CamelCase, and please add a documentation comment for this method.

Reply:
This method was completely removed in favor of throwing the Diagnostic directly.